Bug/#198/add verify button in myicard page when status is inactive

Ccolina03 opened the pull request:
-Added button component when Inactive(Reverify) and iNactive(Verify).
-Button redirects to Verification page
-Need to implement still the box shadow properly

Ccolina03 committed:
Adding Button verification

Ccolina03 committed:
Adding X to upper part

Ccolina03 committed:
Fixing minor issues with button

yismailuofa commented on mobile/src/components/shared/ICardPage.jsx:
```diff
@@ -13,13 +14,15 @@ import {colors} from '../../utilites/Theme';
   stale: colors.darkGray,
   'verifying account': colors.yellow,
   Unlinked: colors.lightGray,
+  'iNactive': colors.red,
+  'Inactive': colors.red
```
Could we rename these two new states to more specific names. I think it would be very confusing to have 3 states called `inactive` with the only difference being capitalization?

yismailuofa commented on mobile/src/components/shared/ICardPage.jsx:
```diff
@@ -33,13 +36,40 @@ const MyICardPage = (props) => {
 
+
+  //creation of two experimental buttons
+  const Button = ({status, onPress}) => {
+    let content1
+    if (status == 'iNactive') {
+      content1 = (
+        <View>
+          <TouchableOpacity style={styles.button} onPress={onPress}>
+         <Text style={styles.buttonText}> Verify account</Text>
+        </TouchableOpacity>
+        </View>
+      )
+    }
+    else if (status == 'Inactive') {
+      content1 = (
+        <View>
+          <TouchableOpacity style={styles.button} onPress={onPress}>
+         <Text style={styles.buttonText}> Reverify account</Text>
+        </TouchableOpacity>
+        </View>
+      )
+    }
+
+    return <Text>{content1} </Text>
+  }
+
```
Since the code is mostly duplicated, you can merge this into one view function and just use 
`a ? 'Verify account' : b 'Reverify account'` in the text field to conditionally render it.

yismailuofa replied:
Also good to name it something like VerificationButton instead of just Button.

yismailuofa commented on mobile/src/components/shared/ICardPage.jsx:
```diff
@@ -64,24 +95,49 @@ const MyICardPage = (props) => {
- 
-      <View style={styles.container} backgroundColor={statusColors[props.status]} marginTop={props.status == 'inactive' || props.status == 'stale' || props.status == 'Unlinked' ? 109 : 235}>
+
+      <View style={styles.container} backgroundColor={statusColors[props.status]} marginTop={props.status == 'inactive' || props.status == 'stale' || props.status == 'Unlinked'||props.status =='iNactive' || props.status == 'Inactive' ? 109 : 235}>
```
Since this condition is getting long now, we should extract it into its own variable.

yismailuofa commented on mobile/src/components/shared/ICardPage.jsx:
```diff
@@ -64,24 +95,49 @@ const MyICardPage = (props) => {
       </View>
+
+      {/* Experimental Position */}
+      <Button onPress={props.verify} status={props.status}/>
```
We should probably make this button conditionally render only if the user needs to verify.

yismailuofa reviewed (changes requested):
Good work, left a few comments. 👍🏼 Could you also add a screenshot in the PR to showcase your changes.

MoyinF reviewed (commented):
Looks good, just make sure to implement the changes Youssef suggested

Ccolina03 committed:
Adding pull request changes, missing 1

Ccolina03 committed:
Finish bug

Ccolina03 commented:
![WhatsApp Image 2023-01-26 at 11 06 24](https://user-images.githubusercontent.com/107608945/214916337-e48858ab-e3d5-4c31-b39e-ad1782500750.jpeg)
![WhatsApp Image 2023-01-26 at 11 06 23](https://user-images.githubusercontent.com/107608945/214916364-f4a570ab-41f3-47ab-8dc5-f02b6efc62b4.jpeg)
![WhatsApp Image 2023-01-26 at 11 06 24](https://user-images.githubusercontent.com/107608945/214916383-13620542-383d-400d-9e60-b0235a1a6d09.jpeg)

-Implemented changes and refactored the previous code a bit.

Ccolina03 committed:
Back to default Unlinked

Ccolina03 committed:
Unlinked

Ccolina03 committed:
Adding Scrolling in Vendors List of Cards for Homepage

yismailuofa commented:
Closing due to newer PR #54
